bpo-33563: Fileinput(bufsize=0) does not emit deprecation warning.

Carreau opened the pull request:
Also duplicate the warn logic for the stacklevel to be correct both when
calling the constructor directly and calling `fileinput.input`


<!-- issue-number: bpo-33563 -->
https://bugs.python.org/issue33563
<!-- /issue-number -->

serhiy-storchaka commented on Lib/fileinput.py:
```diff
@@ -80,7 +80,9 @@
 _state = None
 
-def input(files=None, inplace=False, backup="", bufsize=0,
+_sentinel=object()
```
Add spaces around `=`.

serhiy-storchaka commented on Lib/test/test_fileinput.py:
```diff
@@ -87,28 +87,35 @@ def close(self):
-                remove_tempfiles(t1, t2, t3, t4)
+        bs = None
+        t1 = t2 = t3 = t4 = t5 = t6 = None
+
```
Superfluous empty lines.

serhiy-storchaka commented on Lib/test/test_fileinput.py:
```diff
@@ -87,28 +87,35 @@ def close(self):
 class BufferSizesTests(unittest.TestCase):
     def test_buffer_sizes(self):
         # First, run the tests with default and teeny buffer size.
```
The tests will be not ran with non-default buffer size. Please restore tests.

Carreau replied:
Ok, but that is useless as `bufsize=` is anyway ignored. That is basically running the same exact test twice...

serhiy-storchaka commented on Lib/test/test_fileinput.py:
```diff
@@ -686,7 +702,6 @@ def do_test_call_input(self):
         self.assertIs(files, result.files, "files")
         self.assertIs(inplace, result.inplace, "inplace")
         self.assertIs(backup, result.backup, "backup")
-        self.assertIs(bufsize, result.bufsize, "bufsize")
```
Please restore this assertion.

Carreau replied:
This is testing the Mock, the MockFileInput have a bufsize attribute, the rel FileInput does not. The goal of this test is to explicitly test that `fileinput.input` is forwrding `input(bufsize=...)` to `FileInput(bufsize...)` which this patch does remove in order to print a warnign with the correct stack. I can change to assertIsNot though.

serhiy-storchaka reviewed (changes requested):
This change removes some existing tests. They should be restored. Even if the *bufsize* parameter is deprecated, the code should work with it.

It may be easier to pass `bufsize=fileinput._sentinel` if you want to test the behavior with the default bufsize.

Commit:
bpo-33563: Fileinput(bufsize=0) does not emit deprecation warning.

Also duplicate the warn logic for the stacklevel to be correct both when
calling the constructor directly and calling `fileinput.input`
